cmd/gapit/common.go

```diff
@@ -49,7 +49,14 @@ func (f CommandFilterFlags) commandFilter(ctx context.Context, client service.Se
 		if err != nil {
 			return nil, log.Err(ctx, err, "Failed to load the contexts")
 		}
-		filter.Context = contexts.(*service.Contexts).List[f.Context].ID
+		numContexts := len(contexts.(*service.Contexts).List)
+		// f.Context is an ID, so it must be strictly inferior to numContexts
+		if f.Context < numContexts {
```

Comment:
More nit: It is more idiomatic to check for errors and return, but have the main code flow be error free. For example:

```go
if n := len(contexts.(*service.Contexts).List); f.Context < n {
    return nil, log.Errf(ctx, err, "Context %d is out of range [0..%d]", f.Context, n-1)
}
filter.Context = contexts.(*service.Contexts).List[f.Context].ID
```

Reply:
Thanks for the feedback, applied!

I understand how it feels more idiomatic, although to be honest I find it less readable as the actual test is in the middle of the long if condition line.

Nit: you got the test wrong ;-)

Reply:
Heh, I haven't had my coffee today. </rubbish excuse> :)

> I find it less readable as the actual test is in the middle of the long if condition line.

True, but it is only long because you are doing a cast as well as a field access. This is more a more common pattern:

```go
contextsBoxed, err := client.Get(ctx, p.Contexts().Path(), nil)
if err != nil {
	return nil, log.Err(ctx, err, "Failed to load the contexts")
}
contexts := contextsBoxed.(*service.Contexts)

if n := len(contexts.List); f.Context >= n {
    return nil, log.Errf(ctx, err, "Context %d is out of range [0..%d]", f.Context, n-1)
}
filter.Context = contexts.List[f.Context].ID
```

But that's really nitpicking.

Reply:
Looks better! Applied, with 'boxedContexts' as variable name as it is how most boxed variables are named.